packages/cohere/src/cohere-chat-language-model.ts

```diff
@@ -156,28 +201,28 @@ export class CohereChatLanguageModel implements LanguageModelV1 {
       fetch: this.config.fetch,
     });
 
-    const { chat_history, message, ...rawSettings } = args;
+    const { messages, ...rawSettings } = args;
     const generateId = this.config.generateId;
 
     return {
-      text: response.text,
-      toolCalls: response.tool_calls
-        ? response.tool_calls.map(toolCall => ({
-            toolCallId: generateId(),
-            toolName: toolCall.name,
-            args: JSON.stringify(toolCall.parameters),
+      text: response.message.content?.[0]?.text ?? '',
+      toolCalls: response.message.tool_calls
+        ? response.message.tool_calls.map(toolCall => ({
+            // TODO(shaper): See if we can remove all `generateId` related bits. Main thing is how it's used in streaming?
```

Comment:
Each tool call needs an ID, ideally the one sent by the provider API. If the provider does not send IDs for tool calls, we generate them.

Reply:
Revised to use provider's tool id.